crates/configuration/src/version3/mod.rs

```diff
@@ -182,6 +182,16 @@ pub async fn introspect(
     let relevant_type_representations =
         filter_type_representations(&scalar_types, type_representations);
 
+    // build a list of names to ensure they are unique. We assume scalar type names + composite types are already a unique set.
+    let mut type_names: HashSet<TypeName> = scalar_types
+        .iter()
+        .map(|t| t.clone().into_inner())
+        .collect();
```

Comment:
You could slightly simplify this and avoid cloning

```suggestion
    let mut type_names: HashSet<TypeName> = scalar_types
        .into_iter()
        .map(|t| t.into_inner())
        .collect();
```